core/src/main/java/org/apache/calcite/rel/rules/SubQueryRemoveRule.java

```diff
@@ -696,16 +696,15 @@ private static RexNode rewriteIn(RexSubQuery e, Set<CorrelationId> variablesSet,
         if (variablesSet.isEmpty()) {
           builder.aggregate(builder.groupKey(builder.field("cs")),
               builder.count(false, "c"));
-
-          // sorts input with desc order since we are interested
-          // only in the case when one of the values is true.
-          // When true value is absent then we are interested
-          // only in false value.
-          builder.sortLimit(0, 1,
-              ImmutableList.of(builder.desc(builder.field("cs"))));
         } else {
           builder.distinct();
         }
+        // sorts input with desc order since we are interested
+        // only in the case when one of the values is true.
+        // When true value is absent then we are interested
+        // only in false value.
+        builder.sortLimit(0, 1,
```

Comment:
This includes the `order by cs limit 1` clause in the correlated query case. Without it, multiple rows are returned, as described in the issue.

Reply:
It looks like this was written by @vvysotskyi about 6 years ago, it would be great if he could comment.